rpm/dnf/libdnf_wrapper.cpp

```diff
@@ -0,0 +1,236 @@
+// +build dnf
+
+#include "libdnf_wrapper.h"
+#include <exception>
+#include <string>
+#include <string.h>
+
+const char* newCString(std::string s) {
+    char* msg = new char[s.size()+1];
+    strcpy(msg, s.c_str());
+    return msg;
+}
+
+const char* getErrorMessage(GError* gerr) {
+    if (gerr == nullptr)
+        return newCString("unknown error");
+
+    const char* msg = strdup(gerr->message);
```

Comment:
I think this is going to create a heap problem.
`newCString()` allocates with `new`; `strdup()` allocates with `malloc()`.
When the caller frees the result, they're always freeing with `C.free()`; but memory allocated with `new` should be returned via `delete`.